index.js

```diff
@@ -401,46 +401,64 @@ Cartero.prototype.processMains = function( callback ) {
 		} );
 	} );
 
-	p.on( 'packageCreated', function( newPackage ) {
+	p.addPreProcessor(function( newPackage, done ) {
 		if( newPackage.isParcel ) {
 			_this.parcelsByEntryPoint[ newPackage.mainPath ] = newPackage;
 		}
 
 		_this.packagePathsToIds[ newPackage.path ] = newPackage.id;
 
-		newPackage.addTransform( replaceStringTransform, {
-			find : /url\(\s*[\"\']?([^)\'\"]+)\s*[\"\']?\s*\)/g,
-			replace : function( file, match, theUrl ) {
-				theUrl = theUrl.trim();
+		_this.writeIndividualAssetsToDisk( newPackage, assetTypesToWriteToDisk, function( err ) {
+			if( err ) return _this.emit( 'error', err );
 
-				// absolute urls stay the same.
-				if( theUrl.charAt( 0 ) === '/' ) return match;
-				if( theUrl.indexOf( 'data:' ) === 0 ) return match; // data url, don't mess with this
+			newPackage.addTransform( replaceStringTransform, {
+				find : /url\(\s*[\"\']?([^)\'\"]+)\s*[\"\']?\s*\)/g,
+				replace : function( file, match, theUrl ) {
+					theUrl = theUrl.trim();
+
+					// absolute urls stay the same.
+					if( theUrl.charAt( 0 ) === '/' ) return match;
+					if( theUrl.indexOf( 'data:' ) === 0 ) return match; // data url, don't mess with this
+
+					var assetUrl = path.resolve( path.dirname( file ), theUrl );
+					var newAssetName = _this.assetMap[ assetUrl ];
+
+					if (newAssetName) {
+						// replace the url with the new name
+						theUrl = path.join( path.dirname( theUrl ), newAssetName );
+					} else {
+						// this happen when we have packages that have assets references that are not specified
+						// in the image tag in package.json. It happens in modules like jqueryui
+						log.warn( file, 'Asset reference ' + theUrl + ' cannot be resolved' );
```

Comment:
Interesting case: example2 in the tests was failing because jqueryui has references to assets that are not specified in `image` in `package.json`. LMK if this is a good treatment of the error